### PR TITLE
Split settings across frontend and backend

### LINKED ISSUE #1171: bring back hidden years
Hidden years hidden years hidden years.

Bring back hidden years. 
You have already mostly completed this.
Finish it in your settings slice to backend issue.

### LINKED ISSUE #1177: shuffle some of the user settings to backend
It doesn't make much sense to persist certain user settings in local storage, thus we want you to shuffle this to backend. 
The split will be (feel free to discuss more):
- user persisted (backend)
  - showMarks
  - hiddenYears
- frontend persisted (redux persist)
  - theme
  - showLockedCourses
  - showPastWarnings

Would be good to see it still stored in redux, so maybe on page load, you fetch and overwrite redux with it, and similarly update it when redux gets changed? You can blacklist them from the redux persist. Could be funky on when to remove this though, on login or logout???

**OR BANGER IDEA**
- dont store the backend stuff in redux, but instead create a custom hook that is `useSettings`
- this just returns a memod object of all the settings and their corresponding mutator callbacks... 
- in this hook, you build the settings object from a combo of redux selectors and the query response
- similarly build their corresponding callbacks from a combo of redux reducers and a wrapper of the query mutation function. 
- if the user is not yet logged in (like someone just wants theme setting), just return the user based settings as undefined...

### PR DESCRIPTION
**Pull request by lhvy**
- Resolves #1177 
- Resolves #1171

### CODE REVIEW

File: frontend/src/pages/GraphicalSelector/HowToUse/HowToUse.tsx [hunk 1]
```diff
@@ -11,14 +11,13 @@ import step4Dark from 'assets/GraphicalSelectorHelp/step4-dark.jpg';
...
 import step5Light from 'assets/GraphicalSelectorHelp/step5-light.jpg';
-import type { RootState } from 'config/store';
 import CS from '../common/styles';
 import S from './styles';
 
 const { Title } = Typography;
 
 const HowToUse = () => {
-  const { theme } = useSelector((state: RootState) => state.settings);
+  const { theme } = useTheme();
```
Comment (ollibowers) [inline frontend/src/pages/GraphicalSelector/HowToUse/HowToUse.tsx]:
```suggestion
  const { theme } = useSettings();
```
i think?

**Review by ollibowers — CHANGES_REQUESTED** [review]
happy once fixed, will also wait till i pass this by martin tonight

File: frontend/src/pages/GraphicalSelector/HowToUse/HowToUse.tsx (hunk 1 above)
Comment (lhvy) [reply frontend/src/pages/GraphicalSelector/HowToUse/HowToUse.tsx]: Nice catch!

**Comment by scorpiontornado** [comment]
LGTM 👍

**Comment by matthew-lim-matthew-lim** [comment]
LGTM 👍

**Review by ollibowers — APPROVED** [review]
looks good to me!!! (and martin)

**Pull request merged**